src/main/scala/io/iohk/ethereum/consensus/ethash/difficulty/EthashDifficultyCalculator.scala

```diff
@@ -15,9 +18,15 @@ class EthashDifficultyCalculator(blockchainConfig: BlockchainConfig) extends Dif
   val ConstantinopleRelaxDifficulty: BigInt = 5000000
 
   def calculateDifficulty(blockNumber: BigInt, blockTimestamp: Long, parentHeader: BlockHeader): BigInt = {
```

Comment:
I've been thinking... given this scenario of B1->B2->C3->B4, that we are calculating the difficulty of B4 what would be the difference between:
1. Do as done here that the difficulty of B4 is calculated using the difference in timestamps of B2 and B1
2. Have B4 have the same difficulty as B2

?

Aren't they mostly the same? (if that's so maybe we could go with 2. as it's simpler)

Reply:
Something else that I noticed and I don't like much is that having the `timestampDiff=B2.timestamp-B1.timestamp` means that it doesn't adapt in any way to a how long B4 takes to mine... I was thinking of something like `(B4.timestamp-B2.timestamp)/2` but it assumes that producing a checkpoint block takes the same time to produce as a regular block. Maybe the best alternative for now is just leave one of the above solutions and re-check that after evaluating the integration with OBFT